### ISSUE TITLE
Getting an error in run.py

### ISSUE THREAD

**Issue by harshilishere** [susnato/TriggerWordDetection#1]
This the error:

```
   File "run.py", line 11, in main
     model = load_model("models/trigger word detection model.h5", compile=False)
   File "/home/harshil/Downloads/kerass/try2/lib/python3.6/site-packages/keras/models.py", line 269, in load_model
     model_config = json.loads(model_config.decode('utf-8'))
   AttributeError: 'str' object has no attribute 'decode'
```

Python3 version is 3.6.9
I simply did a git clone and installed the requirements. Then it is giving me this error. 
Can someone please help me with this?

**Comment by susnato** [comment]
This issue is discussed previously in this thread( https://github.com/tensorflow/tensorflow/issues/44467 ) 
Try to use h5py<3.0.0 in my case it is 2.10.0( !pip install h5py==2.10.0)
Sorry for the late reply 
Thanks

**Comment by harshilishere** [comment]
Hey, thanks for responding. The version of h5py is already 2.10.0 and other packages are of the exact version as mentioned in the git, but still i get the same error.

**Comment by harshilishere** [comment]
i created a new venv and re installed all the packages again.

This is my pip list:

> $ pip list
> Package              Version
> -------------------- ------------
> absl-py              0.11.0
> astor                0.8.1
> cached-property      1.5.2
> cffi                 1.14.4
> cycler               0.10.0
> gast                 0.2.2
> google-pasta         0.2.0
> grpcio               1.34.0
> h5py                 3.1.0
> importlib-metadata   3.3.0
> Keras                2.1.6
> Keras-Applications   1.0.8
> Keras-Preprocessing  1.1.2
> kiwisolver           1.3.1
> Markdown             3.3.3
> matplotlib           3.3.3
> numpy                1.19.4
> opt-einsum           3.3.0
> Pillow               8.0.1
> pip                  20.3.3
> pkg-resources        0.0.0
> protobuf             3.14.0
> PyAudio              0.2.11
> pycparser            2.20
> pydub                0.24.1
> pyparsing            2.4.7
> python-dateutil      2.8.1
> PyYAML               5.3.1
> scipy                1.2.1
> setuptools           51.0.0
> six                  1.15.0
> sounddevice          0.4.0
> SoundFile            0.10.3.post1
> tensorboard          1.15.0
> tensorflow           1.15.0
> tensorflow-estimator 1.15.1
> termcolor            1.1.0
> typing-extensions    3.7.4.3
> Werkzeug             1.0.1
> wheel                0.36.2
> wrapt                1.12.1
> zipp                 3.4.0


Yet I get this:


> python3 run.py 
> /home/harshil/Downloads/kerass/try1/lib/python3.6/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
>   warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
> Using TensorFlow backend.
> Traceback (most recent call last):
>   File "run.py", line 30, in <module>
>     main()
>   File "run.py", line 11, in main
>     model = load_model("models/trigger word detection model.h5", compile=False)
>   File "/home/harshil/Downloads/kerass/try1/lib/python3.6/site-packages/keras/models.py", line 269, in load_model
>     model_config = json.loads(model_config.decode('utf-8'))
> AttributeError: 'str' object has no attribute 'decode'

**Comment by harshilishere** [comment]
alright so I  created a new venv and now its running i guess but not working....
![Screenshot from 2020-12-18 09-06-16](https://user-images.githubusercontent.com/61970592/102571755-4f42eb80-4110-11eb-90c8-7250ad45f952.png)

**Comment by harshilishere** [comment]
so I can ensure that live_record.py is working as it records the audio at audio/current_recording.wav. But probably there
s some issue with run.py. As of now that is way out of my understanding...!

**Comment by susnato** [comment]
Actually the error is caused by pyaudio in live_record.py and I found two threads discussing the same error (https://stackoverflow.com/questions/31603555/unknown-pcm-cards-pcm-rear-pyaudio, https://stackoverflow.com/questions/7088672/pyaudio-working-but-spits-out-error-messages-each-time), I am sorry I can't help you with pyaudio errors but can you please try to check if audio/current_recording.wav is recording the right speech or not?And if you can, please unplug all the devices (I meant bluetooth devices) and just plug one mic with jack and then try to use and see if it is working or not (just say "Activate" load enough and check if it is detecting the word or not)

**Comment by harshilishere** [comment]
Thank you @susnato for the response! I unplugged everything and just kept the mic. And yes it does records the bit in audio/current_recording.wav. It records me saying "activate". I also tried to speak in different accents, I also googled "activate" and tried to play the audio from that played the audio from youtube. But nothing actually worked. 

I tried this code in windows: [https://github.com/Tony607/Keras-Trigger-Word](url)

And it was working quiet well in windows but even this code is not working in my ubuntu. IDK what's the issue
